File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.rs
fn main() {
    let n: usize = {
        let mut input = String::new();
        std::io::stdin().read_line(&mut input).unwrap();
        input.trim().parse().unwrap()
    };

    let mut counter = 0;
    let mut i = 666;

    while counter < n {
        let mut tmp = i;
        let mut six_counter = 0;

        while tmp > 0 {
            if tmp % 10 != 6 {
                six_counter = 0;
                tmp /= 10;
                continue;
            }
            six_counter += 1;
            if six_counter == 3 {
                counter += 1;
                break;
            }
            tmp /= 10;
        }
        i += 1;
    }

    println!("{}", i - 1);
}

File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.py
import sys

read = sys.stdin.readline
write = sys.stdout.write


def main():
    n = int(read())
    res = 0
    for i in range(666, 10000667):
        cnt = 0
        tmp = i
        while tmp:
            if tmp % 10 == 6:
                cnt += 1
                if cnt == 3:
                    res += 1
                    if res == n:
                        write(f"{i}\n")
                        return
                    break
            else:
                cnt = 0
            tmp //= 10


main()
